BlueprintUI/Sources/BlueprintView/BlueprintView.swift

```diff
@@ -179,45 +183,109 @@ public final class BlueprintView: UIView {
+
+        return measurement
+    }
+
+    ///
+    /// Measures the size needed to display the view within then given constraining size,
+    /// by measuring the current `element` of the `BlueprintView`.
+    ///
+    /// If you would like to not constrain the measurement in a given axis,
+    /// pass `0.0` or `.greatestFiniteMagnitude` for that axis.
+    ///
+    public override func systemLayoutSizeFitting(
+        _ targetSize: CGSize
+    ) -> CGSize {
+        /// For us, this is the same as `sizeThatFits`, since blueprint does not
+        /// translate the concept of constraints.
+        sizeThatFits(targetSize)
+    }
+
+    ///
+    /// Measures the size needed to display the view within then given constraining size,
+    /// by measuring the current `element` of the `BlueprintView`.
+    ///
+    /// If you would like to not constrain the measurement in a given axis,
+    /// pass `0.0` or `.greatestFiniteMagnitude` for that axis.
+    ///
+    public override func systemLayoutSizeFitting(
+        _ targetSize: CGSize,
+        withHorizontalFittingPriority horizontalFittingPriority: UILayoutPriority,
+        verticalFittingPriority: UILayoutPriority
+    ) -> CGSize {
+        /// For us, this is the same as `sizeThatFits`, since blueprint does not
+        /// translate the concept of horizontal or vertical constraints.
+        sizeThatFits(targetSize)
     }
 
     public override var intrinsicContentSize: CGSize {
-        if let intrinsicContentSize = cachedIntrinsicContentSize {
-            return intrinsicContentSize
+        guard element != nil else {
+            return CGSize(
+                width: UIView.noIntrinsicMetric,
+                height: UIView.noIntrinsicMetric
+            )
         }
 
-        let intrinsicContentSize = resolveIntrinsicContentSize()
-        cachedIntrinsicContentSize = intrinsicContentSize
-        return intrinsicContentSize
-    }
+        let constraint = { () -> SizeConstraint in
+            if bounds.width == 0 {
+                return .unconstrained
+            } else {
+                return .init(width: bounds.width)
+            }
+        }()
 
-    public override func invalidateIntrinsicContentSize() {
-        cachedIntrinsicContentSize = nil
-        super.invalidateIntrinsicContentSize()
+        return sizeThatFits(constraint.maximum)
     }
 
     public override var semanticContentAttribute: UISemanticContentAttribute {
         didSet {
             if semanticContentAttribute != oldValue {
                 setNeedsViewHierarchyUpdate()
+                invalidateCachedSizes()
             }
         }
     }
 
     public override func safeAreaInsetsDidChange() {
         super.safeAreaInsetsDidChange()
+
         setNeedsViewHierarchyUpdate()
+        invalidateCachedSizes()
+    }
+
+    public override var frame: CGRect {
+        didSet {
+            guard oldValue != frame else { return }
+
+            invalidateIntrinsicContentSize()
```

Comment:
I would have thought that UIKit does this automatically; maybe I'm wrong

Reply:
Frame and bounds are actually derived from the auto layout engine when using it, so AFAIK this doesn't happen automatically.

Reply:
I am going to remove this. The docs say: 

> This intrinsic size **must be independent of the content frame**, because there’s no way to dynamically communicate a changed width to the layout system based on a changed height, for example.